source/developer/webapp-component.rst

```diff
@@ -178,3 +178,129 @@ Testing your Component
 The last required piece of building a webapp component is to test it. That can be done using the component testing framework described in this blog post:
 
 https://grundleborg.github.io/posts/react-component-testing-in-mattermost/
+
+Quick guides on how to do component testing
+~~~~~~~~~~~~~~~~~~
+
+1. Match snapshot using default or expected props. Consider global `mm_config` as necessary. Note that while snapshot is convenient, we require not to rely solely on this for every test case as this is easy to overlook by initiating `jest --updateSnapshot` without carefully inspecting the change.
+
+.. code-block:: javascript
+  :linenos:
+
+  test('should match snapshot, not send email notifications', () => {
+      global.window.mm_config.SendEmailNotifications = 'false';
+      const wrapper = shallow(<EmailNotificationSetting {...requiredProps}/>);
+
+      expect(wrapper).toMatchSnapshot();
+  });
+
+2. Add verification to important element.
+
+.. code-block:: javascript
+  :linenos:
+
+  expect(wrapper.find('#emailNotificationImmediately').exists()).toBe(true);
+
+
+3. Simulate event and verify state changes accordingly.
+
+.. code-block:: javascript
+  :linenos:
+
+  test('should pass handleChange', () => {
+      const wrapper = mountWithIntl(<EmailNotificationSetting {...requiredProps}/>);
+      wrapper.find('#emailNotificationImmediately').simulate('change');
+
+      expect(wrapper.state('enableEmail')).toBe('true');
+      expect(wrapper.state('emailInterval')).toBe(30);
+  });
+
+
+4. Ensure that all functions of a component are tested. This can be done via events, state changes or just calling it directly.
+
+.. code-block:: javascript
+  :linenos:
+
+  wrapper.instance().handleExpand();
+
+  expect(newUpdateSection).toBeCalled();
+  expect(newUpdateSection).toHaveBeenCalledTimes(1);
+  expect(newUpdateSection).toBeCalledWith('email');
+
+5. When a function is passed to a component via props, make sure to test it if it gets called for a particular event calls or state changes..
```

Comment:
Not sure whether this should be:
`...make sure to test it if it gets called for a particular event call or state change.`
OR
`...make sure to test it if it gets called for a particular event call or its state changes.`

Reply:
Thanks @lindy65! I'd be happy to know which one is correct. :)

Reply:
I'd be inclined to go for the 2nd option grammatically Saturn but let's wait for JW to weigh in from a dev point of view in case I am misunderstanding the process.